pages/k8s/charm-easyrsa.md

```diff
@@ -0,0 +1,131 @@
+---
+wrapper_template: "kubernetes/docs/base_docs.html"
+markdown_includes:
+  nav: "kubernetes/docs/shared/_side-navigation.md"
+context:
+  title: "EasyRSA charm"
+  description: Kubernetes-master Charm reference
+keywords: kubernetes-master, charm, config
+tags: [reference]
+sidebar: k8smain-sidebar
+permalink: charm-easyrsa.html
+layout: [base, ubuntu-com]
+toc: False
+---
+
+
+
+This charm delivers the EasyRSA application to act as a Certificate Authority
+(CA) and creates certificates for related charms.
+
+EasyRSA is a command line utility to build and manage Public Key
+Infrastructure (PKI) Certificate Authority (CA).
+
+The purpose of a Public Key Infrastructure (PKI) is to facilitate the secure
+electronic transfer of information.
+
+## Deployment
+You can deploy an EasyRSA charm with Juju:
+
+```
+juju deploy easyrsa
+juju deploy tls-client
+juju add-relation easyrsa tls-client
+```
+
+## Using the easyrsa charm
+
+The easyrsa charm will become a Certificate Authority (CA) and generate a CA
+certificate. Other charms need only to relate to easyrsa with a requires
+using the `tls-certificates` interface.
+
+To get a server certificate from easyrsa, the charm must include the
+`interface:tls-certificates` interface in the `layer.yaml` file. The charm must
+also require the `tls` interface, in the `metadata.yaml`. The relation name may
+be named what ever you wish, assume the relation is named "certificates" for
+these examples.
+
+### CA
```

Comment:
Not the sure the code example are appropriate here, but we can leave them for now.